lib/rules/key-spacing.js

```diff
@@ -72,6 +72,27 @@ function isSingleLine(node) {
     return (node.loc.end.line === node.loc.start.line);
 }
 
+/** Sets option values from the configured options with defaults
+ * @param {Object} toOptions Object to be initialized
+ * @param {Object} fromOptions Object to be initialized from
+ * @param {Object} defultOptions Object to be used for default values
+ * @returns {Object} The object with correctly initialized options and values
+ */
+function initOptions(toOptions, fromOptions, defultOptions) {
+    toOptions.mode = fromOptions.mode || defultOptions.mode || "strict";
+    if (typeof fromOptions.beforeColon !== "undefined") {
+        toOptions.beforeColon = +!!fromOptions.beforeColon;
+    } else {
+        toOptions.beforeColon = +!!defultOptions.beforeColon;
+    }
+    if (typeof fromOptions.afterColon !== "undefined") {
+        toOptions.afterColon = +!(fromOptions.afterColon === false);
+    } else {
+        toOptions.afterColon = +!(defultOptions.afterColon === false);
+    }
+    return toOptions;
```

Comment:
Can you make the overriding and defaulting for `beforeColon` and `afterColon` more explicit by splitting them into three steps?

1. Same `if` condition as before, but we don't need any boolean coercion (`typeof` checks it exists, schema verifies it's `true` or `false` if it does exist) or defaulting (that comes in step 3), so it can just be `toOptions.xyzColon = +fromOptions.xyzColon;`.
2. `else if` and see if the option on `defaultOptions` is not `undefined` similar to (1), then `toOptions.xyzColon = +defaultOptions.xyzColon;`
3. `else`, default to an explicit `0` or `1` for `beforeColon` and `afterColon`, respectively.